src/core/stores/VNode.ts

```diff
@@ -297,4 +346,31 @@ export class VNode {
         });
         return this;
     }
+    /**
+     * Traverse all previous/next siblings until `callback` returns true. Return
+     * the last encountered sibling.
+     *
+     * @param direction
+     * @param callback
+     * @param skipRange true to skip range nodes
+     */
+    _walkSiblings(
+        direction: 'next' | 'previous',
+        callback: BrowsingCallback,
+        skipRange = false,
```

Comment:
I feel it can get pretty cumbersome to have to always think of the technical, invisible things in our abstraction, and I want it to be possible to... abstract them away.

Reply:
I understand this feeling, but then there are more stuff that needs to be adapted. You can't use a node `index` without knowing if it isn't off by 1 because of a range node. You can't use the number of children of a node without checking whether it contains a range node. We'll need to make a choice. Do we say "it's part of our representation of the document, everyone has to know it and plan accordingly" or do we say "it's technical, I want to hide them" ? The first choice is the easy one since there is nothing to do (not saying it's a better choice, though.) The second one, however, means that, if you want to do it properly, you'll need to adapt all the functions of the document in such a way that it ignores the ranges (and potentially other technical stuff) or not, otherwise it won't make sense that one particular function ignores the range nodes but not the other. If we want to go down the path of hiding them, then it might be useful having something the likes of this:
```
typescript
const sibling = node.nextSibling; // a node
document.mode = 'full';
const range = node.nextSibling; // a range node

assert sibling === range.nextSibling
assert range === range.prevSibling.nextSibling

document.mode = 'lite';

assert range !== range.prevSibling.nextSibling
assert sibling === range.prevSibling.nextSibling
```